compiler/src/dotty/tools/dotc/classpath/ClassPath.scala

```diff
@@ -23,8 +23,15 @@ trait SourceFileEntry extends ClassRepresentation {
 }
 
 case class PackageName(dottedString: String) {
+  val dirPathTrailingSlashJar: String = FileUtils.dirPathInJar(dottedString) + "/"
+
+  val dirPathTrailingSlash: String =
+    if (java.io.File.separatorChar == '/')
+      dirPathTrailingSlashJar
+    else
+      FileUtils.dirPath(dottedString) + java.io.File.separator
```

Comment:
I don't know if this is too cautious. One justification for the duplication is to save hours of debugging for future maintainers.

WDYT @sjrd ?

Reply:
If this file name is used for *lookup* only, then it can be `/` on all platforms, including Windows. Java can find files that you specify with `/` as if they were `\`.

Where it breaks down is when *listing* files, then comparing their paths to some other data where you always have `/`, or if you `split('/'`). Because Windows will give you `\`, which won't be compared/split correctly.

Reply:
Yes, that is the worry I have. String paths like `a/b/c` on windows are mines: if they are accidentally spliced to form a bigger string path, errors are guaranteed. Currently (before this commit), they already leak to `DirectoryPath`, and works accidentally.